src/test/java/com/jcabi/http/mock/MkAnswerMatchersTest.java

```diff
@@ -77,4 +77,51 @@ public void canMatchHeader() {
             )
         );
     }
+
+    /**
+     * MkAnswerMatchers can match MkAnswer binary body.
+     */
+    @Test
+    public void canMatchBinaryBody() {
+        final byte[] body = this.getAllByteValues();
+        final MkAnswer query = Mockito.mock(MkAnswer.class);
+        Mockito.doReturn(body).when(query).content();
+        MatcherAssert.assertThat(
+            query,
+            MkAnswerMatchers.hasContent(
+                Matchers.is(body)
+            )
+        );
+    }
+
+    /**
+     * MkAnswerMatchers can show mismatch of MkAnswer binary body.
+     */
+    @Test
+    public void canNotMatchBinaryBody() {
+        final byte[] body = new byte[]{0};
+        final MkAnswer query = Mockito.mock(MkAnswer.class);
+        Mockito.doReturn(body).when(query).content();
+        final byte[] mismatch = new byte[]{1};
+        MatcherAssert.assertThat(
+            query,
+            Matchers.not(
+                MkAnswerMatchers.hasContent(
+                    Matchers.is(mismatch)
+                )
+            )
+        );
+    }
+
+    /**
+     * Gets all 256 byte values in an array.
+     * @return An array containing all byte values.
+     */
+    private byte[] getAllByteValues() {
+        final byte[] bytes = new byte[Byte.MAX_VALUE];
+        for (int index = 0; index < bytes.length; index += 1) {
```

Comment:
@mkordas I know that would be good enough. Are you saying I must put `++index` over `index += 1`? Long time ago, but I feel like it didn't like `index++`, odd that it would allow `++index`.